client/Containers/Main.jsx

```diff
@@ -2,19 +2,42 @@ import React, { Component } from 'react';
 import UserDashboard from './UserDashboard.jsx';
 import { connect } from 'react-redux';
 import TestProductList from './TestForProductList.jsx';
+import {
+  HashRouter as Router,
+  Route,
+  Switch,
+  Link,
+  useParams,
+} from 'react-router-dom';
+import AllProducts from './AllProducts';
+import MainNav from './MainNav';
+import FeaturedButton from '../components/styles/FeaturedButton';
 
 class Main extends Component {
   render() {
     const { fetchProductList } = this.props;
     return (
-      <div>
-        <hr />
-        <UserDashboard />
-        <hr className="heavy" />
-        <TestProductList />
-      </div>
+      <Router>
+        <div>
+          <Route component={MainNav} />
+          <Route component={AllProducts} path="/api/product/all" exact />
+          <Route component={UserDashboard} path="/api/login" exact />
```

Comment:
I think we can remove `/api` from these routes

Reply:
I would also maybe call it `/products` and then we can down the road we can use queries at the end (like `/products?page=1&size=6` to hook up the pagination controls. I think this might make our app work with hard refreshes and browser history